klogger/build.gradle

```diff
@@ -0,0 +1,7 @@
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    commonMainApi("org.jetbrains.kotlinx:atomicfu:0.15.1")
```

Comment:
The last time I checked, atomicfu required a plugin to be applied? They have lifted that limitation already?

Reply:
Yes, you are right. It does require a plugin. I was trying to add the dependency but Gradle sync was failing so I tried many approaches and this worked. Let me try again to use the plugin instead.
It's weird that the code is working fine without plugin too 🤔

Reply:
@soywiz the plugin is working now but the older version is required (0.14.4 instead of 0.15.1). Some minor changes will be required as the atomic delegation was introduced in 0.15
The build fails using the latest version as mentioned in Kotlin/kotlinx.atomicfu#171. As mentioned in the issue comments, updating the gradle version to 6.8 works fine

Reply:
Should I update Gradle to 6.8 or use older version of the library?

Reply:
Bumped gradle here: https://github.com/korlibs/korge-next/commit/3a3c76bcf27686a00f83a6c1f8f77e6dcc0e72a4

Is the plugin required in the end? Would that force to apply the plugin to all the libraries and applications using klogger, or it is just required for the library directly depending on it?
Sorry if the question is dummy, but since never used it, I really don't know.

Reply:
It is required by KLogger as the plugin does some bytecode transformations for eg optimizing atomic delegates. It is recommended by the AtomicFu lib to

> Declare atomic variables as private val or internal val. You can use just (public) val, but make sure they are not directly accessed outside of your Kotlin module (outside of the source set). Access to the atomic variable itself shall be encapsulated.

It should not be a problem if we follow this. Users of klogger need not apply plugin from what I can infer as I tried running samples with logging enabled and they were working fine.

Thanks for updating the gradle version. I'll make changes to PR.

Reply:
Nice. When synchronizing to the klogger repo I have to remember to apply the plugin and upgrade the gradle version to 6.8